src/app/components/Root/Root.tsx

```diff
@@ -3,16 +3,47 @@ import './Root.css'
 import { SearchResults, SearchBar } from "../../Recipes";
 import { NavBar } from "../NavBar";
 import { WeatherWidget } from "../../Weather";
+import { ThemeName } from "../../types";
+import { themes } from "../../constants";
+
+export type RootState = {
+    themeName: ThemeName;
+}
 
 export class Root extends React.Component {
+    state: RootState = {
+        themeName: 'dark'
+    };
+
+    toggleTheme = () => {
+        this.setState(() => ({
+            themeName: this.state.themeName === 'light' ? 'dark' : 'light'
+        }))
+    };
+
     public render() {
```

Comment:
`createContext()` creates component "specs", just like defining a component with `extends React.Component`. It's a HOC. This would create a new Provider/Consumer on every render. Instead you want to create them in a static `const` somewhere.

Also, `createContext()` takes a type argument:
```typescript
type ThemeContext = { ... };

const {
    Provider: ThemeProvider,
    Consumer: ThemeConsumer
} = React.createContext<ThemeContext>(defaultThemeContext);
```